### PR TITLE
Ps 5044 5 7 binlog decryption

### PR DESCRIPTION
**Pull request by robgolebiowski**
Still need to make it build on centos + clang formatting.

### CODE REVIEW

File: mysql-test/mysql-test-run.pl [hunk 1]
```diff
@@ -159,7 +159,7 @@
 my $path_vardir_trace;      # Unix formatted opt_vardir for trace files
 
 my $DEFAULT_SUITES =
-"main,sys_vars,binlog,binlog_gtid,binlog_nogtid,federated,gis,rpl,rpl_gtid,rpl_nogtid,innodb,innodb_gis,innodb_fts,innodb_zip,innodb_undo,perfschema,funcs_1,opt_trace,parts,auth_sec,query_rewrite_plugins,gcol,sysschema,test_service_sql_api,json,connection_control,test_services,collations,service_udf_registration,service_sys_var_registration,service_status_var_registration,x,secondary_engine,"
+"main,sys_vars,binlog,binlog_gtid,binlog_nogtid,percona_binlog_5_7_decryption,federated,gis,rpl,rpl_gtid,rpl_nogtid,innodb,innodb_gis,innodb_fts,innodb_zip,innodb_undo,perfschema,funcs_1,opt_trace,parts,auth_sec,query_rewrite_plugins,gcol,sysschema,test_service_sql_api,json,connection_control,test_services,collations,service_udf_registration,service_sys_var_registration,service_status_var_registration,x,secondary_engine,"
```
Comment (laurynas-biveinis) [inline mysql-test/mysql-test-run.pl]: TODO The suite needs a shorter name

File: sql/binlog_reader.cc [hunk 2]
```diff
@@ -114,15 +114,18 @@ bool Binlog_event_data_istream::Decryption_buffer::set_size(
...
+    if (++m_number_of_events_with_half_the_size == 100) {
+      // There were already 101 events in a row, which size was a half of currently
+      // allocated size. This is strong indication that we had occured an
+      // event which was unusually big. Shrink the buffer to half the size.
+      if (resize(m_size / 2)) {
+        return true;
+      }
+      m_number_of_events_with_half_the_size = 0;
+    } else {
+      m_number_of_events_with_half_the_size = 0;
```
Comment (laurynas-biveinis) [inline sql/binlog_reader.cc]: Spurious?

File: sql/event_crypt.h [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
 
-#include "basic_ostream.h"
 #include "binlog_crypt_data.h"
-#include "binlog_event.h"
 #include "my_crypt.h"
-#include "rpl_constants.h"
 
-bool encrypt_event(uint32 offs, const Binlog_crypt_data &crypto, uchar *buf,
-                   uchar *ebuf, size_t buf_len);
+// 5.7 event decrypted
```
Comment (laurynas-biveinis) [inline sql/event_crypt.h]: "Decrypt Percona binlog encryption encrypted-events" or similar

File: sql/log_event.cc [hunk 4]
```diff
@@ -5337,7 +5337,7 @@ void Start_encryption_log_event::print(
   my_b_printf(head, "Encryption scheme: %d", crypto_scheme);
   my_b_printf(head, ", key_version: %d", key_version);
   my_b_printf(head, ", nonce: %s ", nonce_buf);
-  my_b_printf(head, "\n# The rest of the binlog is encrypted!\n");
+  my_b_printf(head, "\n# The rest of the binlog is encrypted with PS 5.7 encryption!\n");
```
Comment (laurynas-biveinis) [inline sql/log_event.cc]: Spell out Percona Server

File: sql/log_event.cc [hunk 5]
```diff
@@ -6708,11 +6708,11 @@ Log_event::enum_skip_reason User_var_log_event::do_shall_skip(
 void Unknown_log_event::print(FILE *,
                               PRINT_EVENT_INFO *print_event_info) const {
   if (print_event_info->short_form) return;
-  if (what != kind::ENCRYPTED) {
+  if (what != kind::ENCRYPTED_WITH_5_7) {
     print_header(&print_event_info->head_cache, print_event_info, false);
     my_b_printf(&print_event_info->head_cache, "\n# %s", "Unknown event\n");
   } else
-    my_b_printf(&print_event_info->head_cache, "\n# %s", "Encrypted event\n");
+    my_b_printf(&print_event_info->head_cache, "\n# %s", "Event encrypted with 5.7 PS binlog encryption\n");
```
Comment (laurynas-biveinis) [inline sql/log_event.cc]: Spell out Percona Server

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Looks great, minor comments only.
One question is, what is the goal with MTR?

**Comment by robgolebiowski** [comment]
https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/52/

**Pull request closed without merging**